### PR TITLE
fix(solana): drop deprecated rpc

### PR DESCRIPTION
**Pull request by TheAncientGoat**
This PR removes calls to the deprecated `getConfirmedSignaturesByAddress2` Solana rpc method. It introduces an assumption that the watcher will always detect a burn on Solana before the signature is pruned - we could argue that if the burn log is present, the signature was confirmed and drop the rpc call entirely; but we do want the "txid" (signature in Solana) for constructing the ren transaction, so we should probably keep the assumption

It will fallback to the old method for deployments that do not yet run solana 0.1.7

### CODE REVIEW

**Comment by TheAncientGoat** [comment]
We have to wait until 1.7 is on Solana Mainnet-Beta; it's still on 1.6 as of now

File: watcher/watcher.go [hunk 1]
```diff
@@ -184,12 +184,19 @@ func (fetcher SolFetcher) FetchBurnLogs(ctx context.Context, from uint64, to uin
...
 			recipient := multichain.RawAddress(data[33 : 33+int(recipientLen)])
 
-			signatures, err := fetcher.client.GetConfirmedSignaturesForAddress2(ctx, programPubk, &solanaRPC.GetConfirmedSignaturesForAddress2Opts{})
+			signatures, err := fetcher.client.GetSignaturesForAddress(ctx, burnLogPubk, &solanaRPC.GetSignaturesForAddressOpts{})
 			if err != nil {
-				resultChan <- BurnLogResult{Error: fmt.Errorf("getting burn log txes: %v", err)}
-				return
+				legacySignatures, err := fetcher.client.GetConfirmedSignaturesForAddress2(ctx, burnLogPubk, &solanaRPC.GetConfirmedSignaturesForAddress2Opts{})
+				if err != nil {
+					resultChan <- BurnLogResult{Error: fmt.Errorf("getting burn log txes: %v", err)}
```
Comment (jazg) [inline watcher/watcher.go:191]: Is it worth printing both errors here?

Comment (jazg) [inline watcher/watcher.go:191]: Is it worth printing both errors here?

**Review by jazg — APPROVED** [review]

File: watcher/watcher.go (hunk 1 above)
Comment (TheAncientGoat) [reply watcher/watcher.go:191]: Good idea, will add

**Review by jazg — APPROVED** [review]

**Pull request merged**